tests/tabs.spec.ts

```diff
@@ -0,0 +1,86 @@
+import { test, expect } from '@playwright/test';
+import { waitForCustomEvent } from './lib/helper.ts'
+
+
+test('Tabs E2E Test', async ({ page }) => {
+  await page.goto('/test.html');
+
+  // Confirm All Tabs Render Correctly
+  const selectedTestTabs = page.getByTestId("Selected Test")
+
+  const tab1 = page.getByRole('button', { name: 'Tab 1' })
+  const tab2 = page.getByRole('button', { name: 'Tab 2' })
+  const tab3 = page.getByRole('button', { name: 'Tab 3' })
+
+  // Check Tabs Buttons Are Visible
+  await expect(selectedTestTabs).toBeVisible()
+
+  await expect(tab1).toBeVisible()
+  await expect(tab2).toBeVisible()
+  await expect(tab3).toBeVisible()
+
+  // Check Tab Buttons Have Correct Text
+  await expect(tab1).toHaveText('Tab 1')
+  await expect(tab2).toHaveText('Tab 2')
+  await expect(tab3).toHaveText('Tab 3')
+
+  // Check Selected Attribute is Correct (Tab 2)
+  await expect(selectedTestTabs).toHaveAttribute("selected", "Tab 2")
+
+  // Check Selected Tab Has Correct Background Color of White (Tab 2)
+  await expect(tab1).not.toHaveClass(/bg-white/)
+  await expect(tab2).toHaveClass(/bg-white/)
+  await expect(tab3).not.toHaveClass(/bg-white/)
```

Comment:
[opt] not sure if it is redundant, but should we also check that the unselected tabs have the correct background color? e.g. `expect(tab1).toHaveClass(/bg-gray/)` or whatever the color is, or maybe we should check that the unselected tabs have the same background color

Reply:
In this case, the background color class for gray (bg-black/20) is on the parent and not on the individual buttons for the tabs so I can't test for gray the same way I'm testing for class bg-white on the selected tab.